main.go

```diff
@@ -4,28 +4,47 @@ import (
 	"fmt"
 	"math/big"
 	"os"
+	"strconv"
 )
 
 var big0 = big.NewInt(0)
 var big1 = big.NewInt(1)
 
-func factorial(x *big.Int) *big.Int {
-	if x.Cmp(big0) == 0 {
-		return big1
+func factorial(x int) *big.Int {
+	vals := make(chan *big.Int, x+1)
+	res := make(chan *big.Int)
```

Comment:
if you don't specify the size of the channel, probably some goroutines will wait (maybe that's the desired behavior)

Reply:
yes that one is intentional. `res` (standing for *result*) makes the final `return <- res` wait till the result is ready.